beacon-chain/p2p/sender.go

```diff
@@ -31,18 +33,20 @@ func (s *Service) Send(ctx context.Context, message interface{}, baseTopic strin
 		return nil, err
 	}
 	// do not encode anything if we are sending a metadata request
-	if baseTopic == RPCMetaDataTopic {
-		return stream, nil
-	}
-
-	if _, err := s.Encoding().EncodeWithMaxLength(stream, message); err != nil {
-		traceutil.AnnotateError(span, err)
-		return nil, err
+	if baseTopic != RPCMetaDataTopic {
+		if _, err := s.Encoding().EncodeWithMaxLength(stream, message); err != nil {
+			traceutil.AnnotateError(span, err)
+			_err := stream.Reset()
+			_ = _err
```

Comment:
This isn't great. Why does stream.Reset() return an error if we aren't going to handle it?

Reply:
It returns an error just in case there might be something useful to return (and as I said, it probably should never have returned an error in the first place). We _can_ just log it, but that won't actually do anything in practice.